c/dynalloc.c

```diff
@@ -1164,6 +1165,274 @@ int dynallocUnallocDatasetByDDName(const DynallocDDName *ddName,
   return dynallocUnallocDatasetByDDNameInternal(ddName, flags, sysRC, sysRSN, FALSE);
 }
 
+/*************************/
+#define FIELD_MAX_LENGTH    1024
+#define REQUEST_MAX_LENGTH  2048
+typedef int EXTF();
+#pragma linkage(EXTF,OS_UPSTACK)
+
+static int allocDataSetOpenString(dataSetRequest *request, char *buffer,
+                             int bufferSize ) {
+  char * ptr;
+  char field[FIELD_MAX_LENGTH];
+  int  status = 0, value;
+  char *baseRequest = "alloc new catalog msg(2) ";
+  char tempString[20] = {0};
+
+  /* setup initial request string */
+  strncpy(buffer, baseRequest, bufferSize);
+
+  /* Initialize the DD/FI field */
+  ptr = (request->ddName == NULL ?  DATASET_DEFAULT_DDNAME: request->ddName);
```

Comment:
Please take a look at the existing functions in this file, they use a programmatic way to pass this information.